### ISSUE TITLE
Can not install sails-mongo

### ISSUE THREAD

**Issue by titovanton** [balderdashy/sails-mongo#390]
Hello, I've got the errors during install sails-mongo on ubuntu/trusty64 with majore nodejs release. Here is my vagrant provision file bellow:
```
# MongoDB
apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv EA312927
echo "deb http://repo.mongodb.org/apt/ubuntu trusty/mongodb-org/3.2 multiverse" |\
tee /etc/apt/sources.list.d/mongodb-org-3.2.list
apt-get update
apt-get upgrade -y
apt-get install -y mongodb-org

# node
apt-get autoremove -y node
apt-get install -y python-software-properties python g++ make
curl -sL https://deb.nodesource.com/setup_4.x | bash -
apt-get install -y nodejs

# sails
npm -g install sails

# sails-mongo
npm -g install sails-mongo
```

And I've got these errors:

```
npm WARN peerDependencies The peer dependency kerberos@~0.0 included from mongodb-core will no
npm WARN peerDependencies longer be automatically installed to fulfill the peerDependency
npm WARN peerDependencies in npm 3+. Your application will need to depend on it explicitly.
|
> kerberos@0.0.19 install /usr/lib/node_modules/sails-mongo/node_modules/mongodb/node_modules/kerberos
> (node-gyp rebuild) || (exit 0)

gyp WARN EACCES user "root" does not have permission to access the dev dir "/root/.node-gyp/4.4.0"
gyp WARN EACCES attempting to reinstall using temporary dev dir "/usr/lib/node_modules/sails-mongo/node_modules/mongodb/node_modules/kerberos/.node-gyp"
make: Entering directory `/usr/lib/node_modules/sails-mongo/node_modules/mongodb/node_modules/kerberos/build'
  CXX(target) Release/obj.target/kerberos/lib/kerberos.o
In file included from ../lib/kerberos.cc:1:0:
../lib/kerberos.h:5:27: fatal error: gssapi/gssapi.h: No such file or directory
 #include <gssapi/gssapi.h>
                           ^
compilation terminated.
make: *** [Release/obj.target/kerberos/lib/kerberos.o] Error 1
make: Leaving directory `/usr/lib/node_modules/sails-mongo/node_modules/mongodb/node_modules/kerberos/build'
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Linux 3.13.0-77-generic
gyp ERR! command "/usr/bin/nodejs" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /usr/lib/node_modules/sails-mongo/node_modules/mongodb/node_modules/kerberos
gyp ERR! node -v v4.4.0
gyp ERR! node-gyp -v v3.3.0
gyp ERR! not ok
```

The same I got when I'm installing locally.

**Comment by sailsbot** [comment]
@titovanton Thanks for posting, we'll take a look as soon as possible.  In the meantime, if you haven’t already, please carefully read the [issue contribution guidelines](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md#ii-issue-contributions) and double-check for any missing information above.  In particular, please ensure that this issue is about a stability or performance bug with a  documented feature; and make sure you’ve included detailed instructions on how to reproduce the bug from a clean install.   Finally, don’t forget to include the version of Node.js you tested with, as well as your version of Sails or Waterline, and of any relevant standalone adapters/generators/hooks.

Thank you!

**Comment by arcoirislabs** [comment]
Hi,
I haven't been able to install the sails-mongo module in one my projects.
OS :- Windows 10
Node :- 4.4.2
Sails :- 0.12.1
Here is the verbose

0 info it worked if it ends with ok
1 verbose cli [ 'C:\\Program Files (x86)\\nodejs\\node.exe',
1 verbose cli   'C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'install',
1 verbose cli   'sails-mongo',
1 verbose cli   '--verbose',
1 verbose cli   '--save' ]
2 info using npm@2.15.0
3 info using node@v4.4.2
4 verbose install initial load of C:\Users\user\work\project\package.json
5 verbose installManyTop reading scoped package data from C:\Users\user\work\project\node_modules\bcrypt\package.json
6 verbose installManyTop reading scoped package data from C:\Users\user\work\project\node_modules\connect-mongo\package.json
7 verbose installManyTop reading scoped package data from C:\Users\user\work\project\node_modules\ejs\package.json
8 verbose installManyTop reading scoped package data from C:\Users\user\work\project\node_modules\grunt\package.json
9 verbose installManyTop reading scoped package data from C:\Users\user\work\project\node_modules\grunt-contrib-clean\package.json
10 verbose installManyTop reading scoped package data from C:\Users\user\work\project\node_modules\grunt-contrib-coffee\package.json
11 verbose installManyTop reading scoped package data from C:\Users\user\work\project\node_modules\grunt-contrib-concat\package.json
12 verbose installManyTop reading scoped package data from C:\Users\user\work\project\node_modules\grunt-contrib-copy\package.json
13 verbose installManyTop reading scoped package data from C:\Users\user\work\project\node_modules\grunt-contrib-cssmin\package.json
14 verbose installManyTop reading scoped package data from C:\Users\user\work\project\node_modules\grunt-contrib-jst\package.json
15 verbose installManyTop reading scoped package data from C:\Users\user\work\project\node_modules\grunt-contrib-less\package.json
16 verbose installManyTop reading scoped package data from C:\Users\user\work\project\node_modules\grunt-contrib-uglify\package.json
17 verbose installManyTop reading scoped package data from C:\Users\user\work\project\node_modules\grunt-contrib-watch\package.json
18 verbose installManyTop reading scoped package data from C:\Users\user\work\project\node_modules\grunt-sails-linker\package.json
19 verbose installManyTop reading scoped package data from C:\Users\user\work\project\node_modules\grunt-sync\package.json
20 verbose installManyTop reading scoped package data from C:\Users\user\work\project\node_modules\include-all\package.json
21 verbose installManyTop reading scoped package data from C:\Users\user\work\project\node_modules\passport\package.json
22 verbose installManyTop reading scoped package data from C:\Users\user\work\project\node_modules\passport-local\package.json
23 verbose installManyTop reading scoped package data from C:\Users\user\work\project\node_modules\rc\package.json
24 verbose installManyTop reading scoped package data from C:\Users\user\work\project\node_modules\sails\package.json
25 verbose installManyTop reading scoped package data from C:\Users\user\work\project\node_modules\sails-disk\package.json
26 verbose installManyTop reading scoped package data from C:\Users\user\work\project\node_modules\sails-hook-autoreload\package.json
27 info package.json grunt@0.4.5 No license field.
28 info package.json grunt-contrib-clean@0.6.0 No license field.
29 info package.json grunt-contrib-coffee@0.13.0 No license field.
30 info package.json grunt-contrib-concat@0.5.1 No license field.
31 info package.json grunt-contrib-copy@0.5.0 No license field.
32 info package.json grunt-contrib-cssmin@0.9.0 No license field.
33 info package.json grunt-contrib-jst@0.6.0 No license field.
34 info package.json grunt-contrib-uglify@0.7.0 No license field.
35 info package.json grunt-contrib-watch@0.5.3 No license field.
36 info package.json grunt-sync@0.2.4 No license field.
37 info package.json passport@0.3.2 No README data
38 info package.json rc@1.0.1 No license field.
39 info package.json passport-local@1.0.0 No license field.
40 verbose readDependencies loading dependencies from C:\Users\user\work\project\package.json
41 silly cache add args [ 'sails-mongo', null ]
42 verbose cache add spec sails-mongo
43 silly cache add parsed spec Result {
43 silly cache add   raw: 'sails-mongo',
43 silly cache add   scope: null,
43 silly cache add   name: 'sails-mongo',
43 silly cache add   rawSpec: '',
43 silly cache add   spec: 'latest',
43 silly cache add   type: 'tag' }
44 silly addNamed sails-mongo@latest
45 verbose addNamed "latest" is being treated as a dist-tag for sails-mongo
46 info addNameTag [ 'sails-mongo', 'latest' ]
47 silly mapToRegistry name sails-mongo
48 silly mapToRegistry using default registry
49 silly mapToRegistry registry https://registry.npmjs.org/
50 silly mapToRegistry data Result {
50 silly mapToRegistry   raw: 'sails-mongo',
50 silly mapToRegistry   scope: null,
50 silly mapToRegistry   name: 'sails-mongo',
50 silly mapToRegistry   rawSpec: '',
50 silly mapToRegistry   spec: 'latest',
50 silly mapToRegistry   type: 'tag' }
51 silly mapToRegistry uri https://registry.npmjs.org/sails-mongo
52 verbose addNameTag registry:https://registry.npmjs.org/sails-mongo not in flight; fetching
53 verbose get https://registry.npmjs.org/sails-mongo not expired, no request
54 silly addNameTag next cb for sails-mongo with tag latest
55 silly addNamed sails-mongo@0.12.0
56 verbose addNamed "0.12.0" is a plain semver version for sails-mongo
57 silly cache afterAdd sails-mongo@0.12.0
58 verbose afterAdd C:\Users\user\AppData\Roaming\npm-cache\sails-mongo\0.12.0\package\package.json not in flight; writing
59 verbose correctMkdir C:\Users\user\AppData\Roaming\npm-cache correctMkdir not in flight; initializing
60 verbose afterAdd C:\Users\user\AppData\Roaming\npm-cache\sails-mongo\0.12.0\package\package.json written
61 silly install resolved [ { name: 'sails-mongo',
61 silly install resolved     version: '0.12.0',
61 silly install resolved     description: 'Mongo DB adapter for Sails.js',
61 silly install resolved     main: './lib/adapter.js',
61 silly install resolved     scripts: { test: 'make test' },
61 silly install resolved     keywords: [ 'mongo', 'orm', 'waterline', 'sails' ],
61 silly install resolved     contributors: [ [Object], [Object], [Object], [Object], [Object] ],
61 silly install resolved     license: 'MIT',
61 silly install resolved     repository:
61 silly install resolved      { type: 'git',
61 silly install resolved        url: 'git://github.com/balderdashy/sails-mongo.git' },
61 silly install resolved     dependencies:
61 silly install resolved      { async: '1.5.2',
61 silly install resolved        lodash: '3.10.1',
61 silly install resolved        mongodb: '2.1.3',
61 silly install resolved        validator: '4.5.1',
61 silly install resolved        'waterline-cursor': '~0.0.6',
61 silly install resolved        'waterline-errors': '~0.10.0' },
61 silly install resolved     devDependencies:
61 silly install resolved      { mocha: '*',
61 silly install resolved        'waterline-adapter-tests': '~0.10.0',
61 silly install resolved        'captains-log': '~0.11.1' },
61 silly install resolved     waterlineAdapter:
61 silly install resolved      { waterlineVersion: '~0.10.0',
61 silly install resolved        interfaces: [Object],
61 silly install resolved        features: [Object] },
61 silly install resolved     gitHead: 'fbde51f3f898c77360313f875bc43e0319743ca8',
61 silly install resolved     bugs: { url: 'https://github.com/balderdashy/sails-mongo/issues' },
61 silly install resolved     homepage: 'https://github.com/balderdashy/sails-mongo#readme',
61 silly install resolved     _id: 'sails-mongo@0.12.0',
61 silly install resolved     _shasum: 'ac4b6058b151b65e2294b87cfff8a5d3cd504ef0',
61 silly install resolved     _from: 'sails-mongo@latest',
61 silly install resolved     _npmVersion: '2.14.7',
61 silly install resolved     _nodeVersion: '4.2.2',
61 silly install resolved     _npmUser: { name: 'particlebanana', email: 'particlebanana@gmail.com' },
61 silly install resolved     dist:
61 silly install resolved      { shasum: 'ac4b6058b151b65e2294b87cfff8a5d3cd504ef0',
61 silly install resolved        tarball: 'https://registry.npmjs.org/sails-mongo/-/sails-mongo-0.12.0.tgz' },
61 silly install resolved     maintainers: [ [Object], [Object], [Object], [Object] ],
61 silly install resolved     directories: {},
61 silly install resolved     _resolved: 'https://registry.npmjs.org/sails-mongo/-/sails-mongo-0.12.0.tgz',
61 silly install resolved     readme: 'ERROR: No README data found!' } ]
62 info install sails-mongo@0.12.0 into C:\Users\user\work\project
63 info installOne sails-mongo@0.12.0
64 verbose installOne of sails-mongo to C:\Users\user\work\project not in flight; installing
65 verbose correctMkdir C:\Users\user\AppData\Roaming\npm-cache\_locks correctMkdir not in flight; initializing
66 verbose lock using C:\Users\user\AppData\Roaming\npm-cache\_locks\sails-mongo-9d397c1a89d021da.lock for C:\Users\user\work\project\node_modules\sails-mongo
67 silly install write writing sails-mongo 0.12.0 to C:\Users\user\work\project\node_modules\sails-mongo
68 verbose unbuild node_modules\sails-mongo
69 silly gentlyRm C:\Users\user\work\project\node_modules\sails-mongo is being purged from base C:\Users\user\work\project
70 verbose gentlyRm don't care about contents; nuking C:\Users\user\work\project\node_modules\sails-mongo
71 verbose tar unpack C:\Users\user\AppData\Roaming\npm-cache\sails-mongo\0.12.0\package.tgz
72 verbose tar unpacking to C:\Users\user\work\project\node_modules\sails-mongo
73 silly gentlyRm C:\Users\user\work\project\node_modules\sails-mongo is being purged
74 verbose gentlyRm don't care about contents; nuking C:\Users\user\work\project\node_modules\sails-mongo
75 silly gunzTarPerm modes [ '777', '666' ]
76 silly gunzTarPerm extractEntry package.json
77 silly gunzTarPerm modified mode [ 'package.json', 420, 438 ]
78 silly gunzTarPerm extractEntry .npmignore
79 silly gunzTarPerm modified mode [ '.npmignore', 420, 438 ]
80 silly gunzTarPerm extractEntry README.md
81 silly gunzTarPerm modified mode [ 'README.md', 420, 438 ]
82 silly gunzTarPerm extractEntry .travis.yml
83 silly gunzTarPerm modified mode [ '.travis.yml', 420, 438 ]
84 silly gunzTarPerm extractEntry CHANGELOG.md
85 silly gunzTarPerm modified mode [ 'CHANGELOG.md', 420, 438 ]
86 silly gunzTarPerm extractEntry Makefile
87 silly gunzTarPerm modified mode [ 'Makefile', 420, 438 ]
88 silly gunzTarPerm extractEntry lib/adapter.js
89 silly gunzTarPerm modified mode [ 'lib/adapter.js', 420, 438 ]
90 silly gunzTarPerm extractEntry lib/collection.js
91 silly gunzTarPerm modified mode [ 'lib/collection.js', 420, 438 ]
92 silly gunzTarPerm extractEntry lib/connection.js
93 silly gunzTarPerm modified mode [ 'lib/connection.js', 420, 438 ]
94 silly gunzTarPerm extractEntry lib/document.js
95 silly gunzTarPerm modified mode [ 'lib/document.js', 420, 438 ]
96 silly gunzTarPerm extractEntry lib/utils.js
97 silly gunzTarPerm modified mode [ 'lib/utils.js', 420, 438 ]
98 silly gunzTarPerm extractEntry lib/query/aggregate.js
99 silly gunzTarPerm modified mode [ 'lib/query/aggregate.js', 420, 438 ]
100 silly gunzTarPerm extractEntry lib/query/index.js
101 silly gunzTarPerm modified mode [ 'lib/query/index.js', 420, 438 ]
102 silly gunzTarPerm extractEntry test/integration/runner.js
103 silly gunzTarPerm modified mode [ 'test/integration/runner.js', 420, 438 ]
104 silly gunzTarPerm extractEntry test/load/loadTest.js
105 silly gunzTarPerm modified mode [ 'test/load/loadTest.js', 420, 438 ]
106 silly gunzTarPerm extractEntry test/support/bootstrap.js
107 silly gunzTarPerm modified mode [ 'test/support/bootstrap.js', 420, 438 ]
108 silly gunzTarPerm extractEntry test/support/config.js
109 silly gunzTarPerm modified mode [ 'test/support/config.js', 420, 438 ]
110 silly gunzTarPerm extractEntry test/support/fixture.js
111 silly gunzTarPerm modified mode [ 'test/support/fixture.js', 420, 438 ]
112 silly gunzTarPerm extractEntry test/unit/adapter.native.js
113 silly gunzTarPerm modified mode [ 'test/unit/adapter.native.js', 420, 438 ]
114 silly gunzTarPerm extractEntry test/unit/adapter.query.OR.test.js
115 silly gunzTarPerm modified mode [ 'test/unit/adapter.query.OR.test.js', 420, 438 ]
116 silly gunzTarPerm extractEntry test/unit/utils.clarifyError.test.js
117 silly gunzTarPerm modified mode [ 'test/unit/utils.clarifyError.test.js', 420, 438 ]
118 silly gunzTarPerm extractEntry test/unit/query/adapter.query.test.js
119 silly gunzTarPerm modified mode [ 'test/unit/query/adapter.query.test.js', 420, 438 ]
120 verbose write writing to C:\Users\user\work\project\node_modules\sails-mongo\package.json
121 info preinstall sails-mongo@0.12.0
122 verbose readDependencies loading dependencies from C:\Users\user\work\project\node_modules\sails-mongo\package.json
123 silly prepareForInstallMany adding async@1.5.2 from sails-mongo dependencies
124 silly prepareForInstallMany adding lodash@3.10.1 from sails-mongo dependencies
125 silly prepareForInstallMany adding mongodb@2.1.3 from sails-mongo dependencies
126 silly prepareForInstallMany adding validator@4.5.1 from sails-mongo dependencies
127 silly prepareForInstallMany adding waterline-cursor@~0.0.6 from sails-mongo dependencies
128 silly prepareForInstallMany adding waterline-errors@~0.10.0 from sails-mongo dependencies
129 verbose readDependencies loading dependencies from C:\Users\user\work\project\node_modules\sails-mongo\package.json
130 silly cache add args [ 'mongodb@2.1.3', null ]
131 verbose cache add spec mongodb@2.1.3
132 silly cache add parsed spec Result {
132 silly cache add   raw: 'mongodb@2.1.3',
132 silly cache add   scope: null,
132 silly cache add   name: 'mongodb',
132 silly cache add   rawSpec: '2.1.3',
132 silly cache add   spec: '2.1.3',
132 silly cache add   type: 'version' }
133 silly addNamed mongodb@2.1.3
134 verbose addNamed "2.1.3" is a plain semver version for mongodb
135 silly mapToRegistry name mongodb
136 silly mapToRegistry using default registry
137 silly mapToRegistry registry https://registry.npmjs.org/
138 silly mapToRegistry data Result {
138 silly mapToRegistry   raw: 'mongodb',
138 silly mapToRegistry   scope: null,
138 silly mapToRegistry   name: 'mongodb',
138 silly mapToRegistry   rawSpec: '',
138 silly mapToRegistry   spec: 'latest',
138 silly mapToRegistry   type: 'tag' }
139 silly mapToRegistry uri https://registry.npmjs.org/mongodb
140 verbose addNameVersion registry:https://registry.npmjs.org/mongodb not in flight; fetching
141 silly cache add args [ 'validator@4.5.1', null ]
142 verbose cache add spec validator@4.5.1
143 silly cache add parsed spec Result {
143 silly cache add   raw: 'validator@4.5.1',
143 silly cache add   scope: null,
143 silly cache add   name: 'validator',
143 silly cache add   rawSpec: '4.5.1',
143 silly cache add   spec: '4.5.1',
143 silly cache add   type: 'version' }
144 silly addNamed validator@4.5.1
145 verbose addNamed "4.5.1" is a plain semver version for validator
146 silly mapToRegistry name validator
147 silly mapToRegistry using default registry
148 silly mapToRegistry registry https://registry.npmjs.org/
149 silly mapToRegistry data Result {
149 silly mapToRegistry   raw: 'validator',
149 silly mapToRegistry   scope: null,
149 silly mapToRegistry   name: 'validator',
149 silly mapToRegistry   rawSpec: '',
149 silly mapToRegistry   spec: 'latest',
149 silly mapToRegistry   type: 'tag' }
150 silly mapToRegistry uri https://registry.npmjs.org/validator
151 verbose addNameVersion registry:https://registry.npmjs.org/validator not in flight; fetching
152 silly cache add args [ 'waterline-cursor@~0.0.6', null ]
153 verbose cache add spec waterline-cursor@~0.0.6
154 silly cache add parsed spec Result {
154 silly cache add   raw: 'waterline-cursor@~0.0.6',
154 silly cache add   scope: null,
154 silly cache add   name: 'waterline-cursor',
154 silly cache add   rawSpec: '~0.0.6',
154 silly cache add   spec: '>=0.0.6 <0.1.0',
154 silly cache add   type: 'range' }
155 silly addNamed waterline-cursor@>=0.0.6 <0.1.0
156 verbose addNamed ">=0.0.6 <0.1.0" is a valid semver range for waterline-cursor
157 silly addNameRange { name: 'waterline-cursor',
157 silly addNameRange   range: '>=0.0.6 <0.1.0',
157 silly addNameRange   hasData: false }
158 silly mapToRegistry name waterline-cursor
159 silly mapToRegistry using default registry
160 silly mapToRegistry registry https://registry.npmjs.org/
161 silly mapToRegistry data Result {
161 silly mapToRegistry   raw: 'waterline-cursor',
161 silly mapToRegistry   scope: null,
161 silly mapToRegistry   name: 'waterline-cursor',
161 silly mapToRegistry   rawSpec: '',
161 silly mapToRegistry   spec: 'latest',
161 silly mapToRegistry   type: 'tag' }
162 silly mapToRegistry uri https://registry.npmjs.org/waterline-cursor
163 verbose addNameRange registry:https://registry.npmjs.org/waterline-cursor not in flight; fetching
164 silly cache add args [ 'waterline-errors@~0.10.0', null ]
165 verbose cache add spec waterline-errors@~0.10.0
166 silly cache add parsed spec Result {
166 silly cache add   raw: 'waterline-errors@~0.10.0',
166 silly cache add   scope: null,
166 silly cache add   name: 'waterline-errors',
166 silly cache add   rawSpec: '~0.10.0',
166 silly cache add   spec: '>=0.10.0 <0.11.0',
166 silly cache add   type: 'range' }
167 silly addNamed waterline-errors@>=0.10.0 <0.11.0
168 verbose addNamed ">=0.10.0 <0.11.0" is a valid semver range for waterline-errors
169 silly addNameRange { name: 'waterline-errors',
169 silly addNameRange   range: '>=0.10.0 <0.11.0',
169 silly addNameRange   hasData: false }
170 silly mapToRegistry name waterline-errors
171 silly mapToRegistry using default registry
172 silly mapToRegistry registry https://registry.npmjs.org/
173 silly mapToRegistry data Result {
173 silly mapToRegistry   raw: 'waterline-errors',
173 silly mapToRegistry   scope: null,
173 silly mapToRegistry   name: 'waterline-errors',
173 silly mapToRegistry   rawSpec: '',
173 silly mapToRegistry   spec: 'latest',
173 silly mapToRegistry   type: 'tag' }
174 silly mapToRegistry uri https://registry.npmjs.org/waterline-errors
175 verbose addNameRange registry:https://registry.npmjs.org/waterline-errors not in flight; fetching
176 silly cache add args [ 'async@1.5.2', null ]
177 verbose cache add spec async@1.5.2
178 silly cache add parsed spec Result {
178 silly cache add   raw: 'async@1.5.2',
178 silly cache add   scope: null,
178 silly cache add   name: 'async',
178 silly cache add   rawSpec: '1.5.2',
178 silly cache add   spec: '1.5.2',
178 silly cache add   type: 'version' }
179 silly addNamed async@1.5.2
180 verbose addNamed "1.5.2" is a plain semver version for async
181 silly mapToRegistry name async
182 silly mapToRegistry using default registry
183 silly mapToRegistry registry https://registry.npmjs.org/
184 silly mapToRegistry data Result {
184 silly mapToRegistry   raw: 'async',
184 silly mapToRegistry   scope: null,
184 silly mapToRegistry   name: 'async',
184 silly mapToRegistry   rawSpec: '',
184 silly mapToRegistry   spec: 'latest',
184 silly mapToRegistry   type: 'tag' }
185 silly mapToRegistry uri https://registry.npmjs.org/async
186 verbose addNameVersion registry:https://registry.npmjs.org/async not in flight; fetching
187 silly cache add args [ 'lodash@3.10.1', null ]
188 verbose cache add spec lodash@3.10.1
189 silly cache add parsed spec Result {
189 silly cache add   raw: 'lodash@3.10.1',
189 silly cache add   scope: null,
189 silly cache add   name: 'lodash',
189 silly cache add   rawSpec: '3.10.1',
189 silly cache add   spec: '3.10.1',
189 silly cache add   type: 'version' }
190 silly addNamed lodash@3.10.1
191 verbose addNamed "3.10.1" is a plain semver version for lodash
192 silly mapToRegistry name lodash
193 silly mapToRegistry using default registry
194 silly mapToRegistry registry https://registry.npmjs.org/
195 silly mapToRegistry data Result {
195 silly mapToRegistry   raw: 'lodash',
195 silly mapToRegistry   scope: null,
195 silly mapToRegistry   name: 'lodash',
195 silly mapToRegistry   rawSpec: '',
195 silly mapToRegistry   spec: 'latest',
195 silly mapToRegistry   type: 'tag' }
196 silly mapToRegistry uri https://registry.npmjs.org/lodash
197 verbose addNameVersion registry:https://registry.npmjs.org/lodash not in flight; fetching
198 verbose get https://registry.npmjs.org/waterline-cursor not expired, no request
199 silly addNameRange number 2 { name: 'waterline-cursor',
199 silly addNameRange   range: '>=0.0.6 <0.1.0',
199 silly addNameRange   hasData: true }
200 silly addNameRange versions [ 'waterline-cursor',
200 silly addNameRange   [ '0.0.0', '0.0.2', '0.0.3', '0.0.4', '0.0.5', '0.0.6' ] ]
201 silly addNamed waterline-cursor@0.0.6
202 verbose addNamed "0.0.6" is a plain semver version for waterline-cursor
203 verbose get https://registry.npmjs.org/mongodb not expired, no request
204 verbose get https://registry.npmjs.org/async not expired, no request
205 verbose get https://registry.npmjs.org/waterline-errors not expired, no request
206 silly addNameRange number 2 { name: 'waterline-errors',
206 silly addNameRange   range: '>=0.10.0 <0.11.0',
206 silly addNameRange   hasData: true }
207 silly addNameRange versions [ 'waterline-errors',
207 silly addNameRange   [ '0.9.0', '0.9.1', '0.10.0-rc1', '0.10.1' ] ]
208 silly addNamed waterline-errors@0.10.1
209 verbose addNamed "0.10.1" is a plain semver version for waterline-errors
210 verbose get https://registry.npmjs.org/lodash not expired, no request
211 verbose get https://registry.npmjs.org/validator not expired, no request
212 silly cache afterAdd waterline-cursor@0.0.6
213 verbose afterAdd C:\Users\user\AppData\Roaming\npm-cache\waterline-cursor\0.0.6\package\package.json not in flight; writing
214 verbose correctMkdir C:\Users\user\AppData\Roaming\npm-cache correctMkdir not in flight; initializing
215 silly cache afterAdd async@1.5.2
216 verbose afterAdd C:\Users\user\AppData\Roaming\npm-cache\async\1.5.2\package\package.json not in flight; writing
217 verbose correctMkdir C:\Users\user\AppData\Roaming\npm-cache correctMkdir not in flight; initializing
218 silly cache afterAdd waterline-errors@0.10.1
219 verbose afterAdd C:\Users\user\AppData\Roaming\npm-cache\waterline-errors\0.10.1\package\package.json not in flight; writing
220 verbose correctMkdir C:\Users\user\AppData\Roaming\npm-cache correctMkdir not in flight; initializing
221 silly cache afterAdd lodash@3.10.1
222 verbose afterAdd C:\Users\user\AppData\Roaming\npm-cache\lodash\3.10.1\package\package.json not in flight; writing
223 verbose correctMkdir C:\Users\user\AppData\Roaming\npm-cache correctMkdir not in flight; initializing
224 verbose afterAdd C:\Users\user\AppData\Roaming\npm-cache\waterline-cursor\0.0.6\package\package.json written
225 verbose afterAdd C:\Users\user\AppData\Roaming\npm-cache\async\1.5.2\package\package.json written
226 verbose afterAdd C:\Users\user\AppData\Roaming\npm-cache\waterline-errors\0.10.1\package\package.json written
227 verbose afterAdd C:\Users\user\AppData\Roaming\npm-cache\lodash\3.10.1\package\package.json written
228 verbose about to build C:\Users\user\work\project\node_modules\sails-mongo
229 verbose unlock done using C:\Users\user\AppData\Roaming\npm-cache\_locks\sails-mongo-9d397c1a89d021da.lock for C:\Users\user\work\project\node_modules\sails-mongo
230 verbose stack Error: version not found: mongodb@2.1.3
230 verbose stack     at setData (C:\Program Files (x86)\nodejs\node_modules\npm\lib\cache\add-named.js:137:12)
230 verbose stack     at RES (C:\Program Files (x86)\nodejs\node_modules\npm\node_modules\inflight\inflight.js:23:14)
230 verbose stack     at f (C:\Program Files (x86)\nodejs\node_modules\npm\node_modules\once\once.js:17:25)
230 verbose stack     at fixName (C:\Program Files (x86)\nodejs\node_modules\npm\lib\cache\add-named.js:29:5)
230 verbose stack     at CachingRegistryClient.get_ (C:\Program Files (x86)\nodejs\node_modules\npm\lib\cache\caching-client.js:125:16)
230 verbose stack     at C:\Program Files (x86)\nodejs\node_modules\npm\lib\cache\caching-client.js:91:14
230 verbose stack     at C:\Program Files (x86)\nodejs\node_modules\npm\node_modules\graceful-fs\graceful-fs.js:78:16
230 verbose stack     at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:380:3)
231 verbose statusCode 404
232 verbose cwd C:\Users\user\work\project
233 error Windows_NT 10.0.10586
234 error argv "C:\\Program Files (x86)\\nodejs\\node.exe" "C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install" "sails-mongo" "--verbose" "--save"
235 error node v4.4.2
236 error npm  v2.15.0
237 error version not found: mongodb@2.1.3
238 error If you need help, you may report this error at:
238 error     <https://github.com/npm/npm/issues>
239 verbose exit [ 1, true ]
240 verbose unbuild node_modules\sails-mongo
241 info preuninstall sails-mongo@0.12.0
242 info uninstall sails-mongo@0.12.0
243 verbose unbuild rmStuff sails-mongo@0.12.0 from C:\Users\user\work\project\node_modules
244 info postuninstall sails-mongo@0.12.0
245 silly gentlyRm C:\Users\user\work\project\node_modules\sails-mongo is being purged from base C:\Users\user\work\project
246 verbose gentlyRm don't care about contents; nuking C:\Users\user\work\project\node_modules\sails-mongo
247 silly vacuum-fs purging C:\Users\user\work\project\node_modules\sails-mongo
248 silly vacuum-fs quitting because other entries in C:\Users\user\work\project\node_modules

Is this a npm issue or something is missing in my system?
Thanks

**Comment by particlebanana** [comment]
> error version not found: mongodb@2.1.3

Hmm try running npm install again. `mongodb@2.1.3` is definitely in the registry.

**Comment by arcoirislabs** [comment]
@particlebanana I tried numerous times. Still not working. I even tried to install the 0.11.7 version still getting the same error at validator package.

**Comment by westlakem** [comment]
any updates on this?  I had to go back to the 0.10.9 release in order to get it to install properly

**Comment by arcoirislabs** [comment]
The problem was absence of .NET framework. Once we installed Visual Studio, we were able to install the module

**Comment by westlakem** [comment]
@arcoirislabs not exactly an option on linux OS

**Comment by arcoirislabs** [comment]
@westlakem Well this error was observed on Windows. On Linux, if this error occurs, it could be due to absence of mongodb and build-essentail packages

**Comment by westlakem** [comment]
@arcoirislabs that's an interesting theory, although the original issue was detected on ubuntu/trusty64, and that was the issue I was referring to.

**Comment by sgress454** [comment]
@westlakem It seems like the problem is installing the `mongodb` module.  Can you try `npm install mongodb@2.1.3` and see what happens?  Also, what version of node / NPM are you using?

**Comment by arcoirislabs** [comment]
@westlakem @sgress454 It has been observed that the kerberos module, that mongodb module depends upon, needs necessary environment based compilers to compile itself. Incomplete compiler packages installation or presence may lead to npm errors during sails-mongo installation. If the inner layer dependency like kerberos is not resolved, it will affect the outer layer modules who have a cyclic dependency. I think to install any mongo based module on Linux you will need the build-essential package to fill in the unsupplied compilers void that is present in various linux based systems. And on windows, it might be necessary to have .NET 3 & 4 along with Python & C++ compiler for Python to complete the installation with no errors.

**Comment by sailsbot** [comment]
@titovanton,@sailsbot,@arcoirislabs,@particlebanana,@westlakem,@sgress454: Hello, I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message and simply close the issue if possible.  On the other hand, if you are still waiting on a patch, please post a comment to keep the thread alive (with any new information you can provide).



If no further activity occurs on this thread within the next 3 days, the issue will automatically be closed.

Thanks so much for your help!

**Comment by westlakem** [comment]
@arcoirislabs Do you recommend reaching out to kerberos, or has @sgress454 already done that?
